boot/zephyr/flash_map.c

```diff
@@ -31,28 +31,45 @@
 
 extern struct device *boot_flash_device;
 
+/* For now, we only support one flash device. */
+#define FLASH_DEVICE_ID 0
```

Comment:
I wonder if we should use something non-zero, and unlikely to conflict with something else. That way we would catch instances where different APIs have different ids that are used for different purposes.